src/core/Persister/Serialization/ProtoToStorageConverter.cpp

```diff
@@ -0,0 +1,27 @@
+//
+// Created by rodion on 9/27/20.
+//
+
+#include "ProtoToStorageConverter.h"
+#include "ProtoToTaskConverter.h"
+
+std::unique_ptr<FullStorage> ProtoToStorageConverter::ConvertFromProto(const StorageProto &storage_proto) {
+  auto temp_storage_ = std::make_unique<FullStorage>();
+//  FullStorage temp_storage_;
+  for(const auto& task : storage_proto.tasks()) {
+    auto task_ = ProtoToTaskConverter::UnconvertTask(task);
+    if(!task_.has_value()) return nullptr;
```

Comment:
Why don't you use std::optional? 
If you are returning nullptr, this is better to use std::nulloptr.
Also it will be readable. 
What is better: check for nullptr or has_value ?